lib/net/nforceif/src/sys_arch.c

```diff
@@ -372,6 +313,25 @@ sys_arch_sem_wait(struct sys_sem **s, u32_t timeout)
     return 0;
 }
 
+int sys_arch_sem_try(struct sys_sem **s)
+{
+    struct sys_sem *sem;
+    LWIP_ASSERT("invalid sem", (s != NULL) && (*s != NULL));
+    sem = *s;
+
+    LARGE_INTEGER timeout;
+    timeout.QuadPart = 0;
+
+    NTSTATUS status = NtWaitForSingleObject(sem->handle, FALSE, &timeout);
+    if (NT_SUCCESS(status)) {
+        if (status == STATUS_TIMEOUT) {
+            return 0;
+        }
+    }
+
+    return NT_SUCCESS(status);
```

Comment:
nit: I feel this is a bit ugly. I prefer:

```c
    if (NT_SUCCESS(status)) {
        if (status == STATUS_TIMEOUT) {
            return 0;
        }
        return 1;
    } else {
        return 0;
    }
```

or

```c
    if (!NT_SUCCESS(status)) {
        return 0;
    }
   
    if (status == STATUS_TIMEOUT) {
        return 0;
    }

    return 1;
```

Reply:
There's no need for exact return values here, so I'd rather avoid additional comparison logic.

Reply:
But your code does more checks? (Your checking logic starts 5 lines up from where I left my comment)

Your logic checks `NT_SUCCESS` twice, and then it's not very clear if `NT_SUCCESS` is just 0 or 1, or if it might actually just forward positive values (which might not be intended).

Reply:
Hm, ok, I changed it to the second variant (using `-1` instead of `1` though).